### PR TITLE
Add my enumerable methods

### PR DESCRIPTION
**Pull request by fatymahmed**

### CODE REVIEW

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
Hi @fatymahmed  good start on this, but we have to correct some things, I'll be pointing a few out explicitly, others will be stated you would find the fix easy anyways

File: my_methods.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(i, self[i])
+      i += 1
+    end
+  end
```
Comment (Oghenebrume50) [inline my_methods.rb:21]: pay attention the order of the arguments

File: my_methods.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  def my_select
+    result = []
+    my_each { |element| result << element if yield(element) }
+  end
```
Comment (Oghenebrume50) [inline my_methods.rb:27]: This function should return something

File: my_methods.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    flag = false
+    my_each { |element| flag ||= yield(element) }
+    flag
+  end
+
+  def my_none
+    flag = false
+    my_each { |element| flag ||= yield(element) }
+    !flag
+  end
```
Comment (Oghenebrume50) [inline my_methods.rb:44]: my_all, my_any, my_none do not work properly, please check them out.

File: my_methods.rb [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+  def my_count
+    count = 0
+    my_each { |element| count += 1 if yield(element) }
+    count
+  end
```
Comment (Oghenebrume50) [inline my_methods.rb:55]: the functionality here is limited, it should work with an argument passed just like the main count method. Please check other functionalities

File: my_methods.rb [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+  def my_inject(initial)
+    result = initial || 0
+    my_each do |element|
+      result = yield(result, element)
+    end
+    result
+  end
+end
```
Comment (Oghenebrume50) [inline my_methods.rb:77]: You have to think of other ways to implement my_map and my_inject, my_each has no return value so it cannot send anything to the array.

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
Please work on those changes test them thoroughly and submit for re-review, kudos on using Gitflow. :+1:

File: my_methods.rb [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+    result
+  end
+
+  def my_all
+    my_each do |item|
+        unless !yield(item)
+          return false
+      end
+    true
+  end
```
Comment (maelfosso) [inline my_methods.rb:33]: Not working

```
>> %w[ant bear cat].all? { |word| word.length >= 3 }
=> true
>> %w[ant bear cat].my_all { |word| word.length >= 3 }
=> false
```

File: my_methods.rb [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+  def my_count(arg = nil)
+    count = 0
+    if arg.nil?
+      my_each { |element| count += 1 if yield(element) }
+    else
+      my_each { |element| count += 1 if element == arg }
+    end
+    count
+  end
```
Comment (maelfosso) [inline my_methods.rb:55]: what you did is limited, take a look here
```
=> [1, 2, 4, 2]
>> ary.count
=> 4
>> ary.my_count
Traceback (most recent call last):
        9: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `<main>'
        8: from /home/maelfosso/.rvm/gems/ruby-2.6.0/bin/ruby_executable_hooks:24:in `eval'
        7: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `<main>'
        6: from /usr/share/rvm/rubies/ruby-2.6.0/bin/irb:23:in `load'
        5: from /usr/share/rvm/rubies/ruby-2.6.0/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        4: from (irb):17
        3: from my_methods.rb:57:in `my_count'
        2: from my_methods.rb:9:in `my_each'
        1: from my_methods.rb:57:in `block in my_count'
LocalJumpError (no block given (yield))
```

**Comment by maelfosso** [comment]
Hello,

You did a really nice job. But there are some changes you should do

- [x] The code I downloaded has some errors. To check out whether you ruby file have some errors, you could run `ruby -wc my_methods.rb`. The results I obtain are

```
$ ruby -wc my_methods.rb
my_methods.rb:33: warning: mismatched indentations at 'end' with 'unless' at 31
my_methods.rb:82: warning: mismatched indentations at 'end' with 'def' at 29
my_methods.rb:112: syntax error, unexpected end-of-input, expecting end
```

- [x] There are remarks about your code. Take a look at them, please.

Kindly make those changes and submit a new code review request.

**Review by bellom — APPROVED** [review]
Hi @fatymahmed 
Great work making all the necessary corrections from the previous review...
## Congratulation, and make sure to use Git Flow in your next project

**Pull request merged**